### PR TITLE
Update blackduck-common and check Jira permissions

### PR DESCRIPTION
**Pull request by gkillough**

### CODE REVIEW

File: src/main/java/com/synopsys/integration/alert/channel/jira/cloud/actions/JiraCloudGlobalTestAction.java [hunk 1]
```diff
@@ -76,6 +79,16 @@ protected boolean isUserMissing(FieldAccessor fieldAccessor) throws IntegrationE
...
+    @Override
+    protected boolean isUserAdmin(FieldAccessor fieldAccessor) throws IntegrationException {
+        JiraCloudProperties jiraProperties = createJiraProperties(fieldAccessor);
+        JiraCloudServiceFactory jiraCloudServiceFactory = jiraProperties.createJiraServicesCloudFactory(logger, gson);
+        MyPermissionsService myPermissionsService = jiraCloudServiceFactory.createMyPermissionsService();
+        MultiPermissionResponseModel myPermissions = myPermissionsService.getMyPermissions(JiraGlobalTestAction.JIRA_ADMIN_PERMISSION_NAME);
+        PermissionModel adminPermission = myPermissions.extractPermission(JiraGlobalTestAction.JIRA_ADMIN_PERMISSION_NAME);
+        return null != adminPermission && adminPermission.getHavePermission();
+    }
+
```
Comment (gkillough) [inline src/main/java/com/synopsys/integration/alert/channel/jira/cloud/actions/JiraCloudGlobalTestAction.java:91]: This chunk is the actual feature. The rest is resolving the dependency hell from blackduck-common.

File: src/main/java/com/synopsys/integration/alert/channel/jira/common/JiraGlobalTestAction.java [hunk 2]
```diff
@@ -45,8 +49,14 @@ public MessageResult testConfig(String configId, FieldModel fieldModel, FieldAcc
...
 
-            if (isAppCheckEnabled(registeredFieldValues) && isAppMissing(registeredFieldValues)) {
-                throw new AlertException(String.format("Please configure the %s plugin for your server.", getChannelDisplayName()));
+            if (isAppCheckEnabled(registeredFieldValues)) {
+                if (isUserAdmin(registeredFieldValues)) {
+                    throw new AlertException("The configured user must be an admin if 'Plugin Check' is enabled");
+                }
+
+                if (isAppMissing(registeredFieldValues)) {
+                    throw new AlertException(String.format("Please configure the %s plugin for your server.", getChannelDisplayName()));
```
Comment (jamesrichard91) [inline src/main/java/com/synopsys/integration/alert/channel/jira/common/JiraGlobalTestAction.java:58]: I think this exception is misleading, I don't think using the channel display name is clear enough. 

We probably want something like this `Please configure the 'Alert Issue Property Indexer' app for you server.` That way they know exactly the name of the app to look for in the marketplace.

File: src/main/java/com/synopsys/integration/alert/channel/jira/server/web/JiraServerCustomFunctionAction.java [hunk 3]
```diff
@@ -76,15 +76,15 @@ public JiraServerCustomFunctionAction(AuthorizationManager authorizationManager,
...
                 if (RestConstants.NOT_FOUND_404 == e.getHttpStatusCode()) {
                     return new ActionResponse<>(HttpStatus.NOT_FOUND,
                         "The marketplace listing of the Alert Issue Property Indexer app may not support your version of Jira. Please install the app manually or request a compatibility update. Error: " + e.getMessage());
                 }
                 createBadRequestIntegrationException(e);
             }
-            boolean jiraPluginInstalled = JiraPluginCheckUtil.checkIsAppInstalledAndRetryIfNecessary(jiraAppService, username, password);
+            boolean jiraPluginInstalled = JiraPluginCheckUtil.checkIsAppInstalledAndRetryIfNecessary(jiraAppService);
             if (!jiraPluginInstalled) {
                 return new ActionResponse<>(HttpStatus.NOT_FOUND, "Was not able to confirm Jira server successfully installed the Jira Server plugin. Please verify the installation on you Jira server.");
```
Comment (jamesrichard91) [inline src/main/java/com/synopsys/integration/alert/channel/jira/server/web/JiraServerCustomFunctionAction.java:89]: Same issue as my previous comment. We probably want to update this message with the name of the App so they know what to look for in the installed apps, and if it isn't there then they know what to search for in the marketplace.

File: src/main/java/com/synopsys/integration/alert/channel/jira/common/JiraGlobalTestAction.java (hunk 2 above)
Comment (gkillough) [reply src/main/java/com/synopsys/integration/alert/channel/jira/common/JiraGlobalTestAction.java:58]: Good idea. This text predated the current PR. It was just moved to its own if.

File: src/main/java/com/synopsys/integration/alert/channel/jira/cloud/web/JiraCloudCustomEndpoint.java [hunk 4]
```diff
@@ -73,25 +72,24 @@ public JiraCloudCustomEndpoint(AuthorizationManager authorizationManager, JiraCl
...
+            if (!HttpStatusCodes.isSuccess(statusCode)) {
+                return new ActionResponse<>(HttpStatus.BAD_REQUEST, "The Jira Cloud server responded with error code: " + statusCode);
             }
 
-            boolean jiraPluginInstalled = JiraPluginCheckUtil.checkIsAppInstalledAndRetryIfNecessary(jiraAppService, username, accessToken);
+            boolean jiraPluginInstalled = JiraPluginCheckUtil.checkIsAppInstalledAndRetryIfNecessary(jiraAppService);
             if (!jiraPluginInstalled) {
-                return new ActionResponse<>(HttpStatus.NOT_FOUND, "Unable to confirm Jira Cloud successfully installed the Jira Cloud plugin. Please verify the installation on you Jira Cloud server.");
+                return new ActionResponse<>(HttpStatus.NOT_FOUND, String.format(
+                    "Unable to confirm successful installation of the Jira Cloud '%s' plugin. Please verify the installation on you Jira Cloud server.", JiraConstants.JIRA_ALERT_APP_NAME));
```
Comment (psantos1113) [inline src/main/java/com/synopsys/integration/alert/channel/jira/cloud/web/JiraCloudCustomEndpoint.java:83]: Existing issue but "Please verify the installation on **you** Jira Cloud server."  Should change to "Please verify the installation on **your** Jira Cloud server."

**Pull request merged**